sdk/storage/azblob/service.go

```diff
@@ -15,32 +15,78 @@ const (
 	scope = "https://storage.azure.com/.default"
 )
 
+// ServiceVersion enumerates the possible service API versions.
+type ServiceVersion string
+
+const (
+	// V2019_02_02 specifies the 2019-02-02 service API version.
+	V2019_02_02 = "2019-02-02"
+)
+
+//ServiceClient provides access to Azure Storage service resources and blob containers.
 type ServiceClient struct {
 	u *url.URL
 	p azcore.Pipeline
+	o ServiceClientOptions
+}
+
+// ServiceClientOptions contains client configuration options for connecting to blob storage.
+type ServiceClientOptions struct {
+	azcore.PipelineOptions
+	// Version is the service API version to use when making requests.
+	// Leaving this empty will default to the latest API version.
+	Version ServiceVersion
+
+	// SecondaryEndpoint is the secondary storage endpoint that can be read from if the storage
+	// account is enabled for RA-GRS.
+	// If this value is set, the secondary endpoint will be used for GET or HEAD requests during retries.
+	// If the status of the response from the secondary endpoint is a 404, then subsequent retries for
+	// the request will not use the secondary endpoint again, as this indicates that the resource
+	// may not have propagated there yet. Otherwise, subsequent retries will alternate back and forth
+	// between the primary and secondary endpoint.
+	SecondaryEndpoint string
```

Comment:
In Track 1.8, this is a retry policy option and I think that is where this belongs; not here. Why was this moved here?